### ISSUE TITLE
Feedback on the sidebar

### ISSUE THREAD

**Issue by RinzeVanDerBrug** [Sterc/modx-redesign#1]
We're looking for feedback on the left sidebar (tree)

https://github.com/Sterc/modx-redesign/blob/master/MODX%20-%20Edit%20page%20-%20Switch%20on.png

![modx - edit page - switch on](https://user-images.githubusercontent.com/381017/28659621-686f973e-72b0-11e7-9ba0-10262c6f4699.png)

We changed from horizontal tabs to vertical tabs, in order to make it more accessible and mobile-friendly. It also allows us to expand it easier. E.g. extras can their own vertical tabs there.

The prototype which is in there now though, becomes a bit messy somehow. Ideas are welcome.

**Comment by rtripault** [comment]
I guess to make it visually "lighter" (and probably DOM wise) you could remove the first "wrapper" (with title "your modx tree structure")

Edit : FWIW, here is my attempt to handle that sidebar : https://youtu.be/3CZJHPcr3bY (working prototype)

**Comment by RinzeVanDerBrug** [comment]
Very interesting. A mash-up between these two could probably lead to an interesting result. The content you added to your tree: it's more then the original resources / media / files. How did you decide on what additionally information to put in there?

**Comment by rtripault** [comment]
It's dynamically generated depending on the user (at least that's the idea).

Using my account ("admin/site builder") i need to access more data than a content editor (like snippets/chunks and the like).

My feeling is the navigation should be custom tailored to the user needs.

Hoping that answers your question

Edit : while probably not obvious in that video, each original tree can be selectively added as a menu entry. For example, let's say i just want to pick a single context (in a multi contexts setup) i can do so. Likewise for the elements and media sources

**Comment by RinzeVanDerBrug** [comment]
Total agreement on the custom tailoring part. 

Did you merge the top navigation and the sidebar?

**Comment by hugopeek** [comment]
Hi Rinze, great that you started working on this!

I agree with you in that the sidebar gets a little messy. That was my first thought actually when I saw the design: too much wrapping going on there. It somehow confuses the eye, but on the other hand you also need to have some visual cues that things are nested somehow.. Difficult buggars, these accordion menus ;)

It will also become annoying on narrower screens I guess, because the 3 layers worth of borders take up quite some pixels there.. @rtripault's comment of removing the outermost wrapper is already a good one.

But do you think it's possible to create a sidebar that completely stretches from the top menu to the bottom of the viewport? And docks to the left side, like Romains example also? That will set it apart from the main area so much better, and buy you some extra space as well..

Further thoughts:

- Good choice collection the various "New ..." buttons under 1 clear Add button
- The "Your resources" on the left there is redundant though, since it says "Resources" right above it ;)
- Quick question: are right mouse click actions still there in this menu?
- Maybe you can accentuate the active context a little also, with a slightly darker grey or something?
- Maybe the refresh wheels in all the tab headings can be visible only on mouseover too.. I like that the menu is relatively clean now, in terms of icons and visual elements (let's indeed stop turning every UI into a FontAwesome christmas tree), but the refresh wheels are still a bit "present"

Can't think of anything else now. Unless you want feedback on the rest too..

**Comment by RinzeVanDerBrug** [comment]
@hugopeek Thanks! Great feedback. In an earlier version I tried to stretch the tree from navigation to bottom, but it felt a little stiff IMO:
![new-interface-display](https://user-images.githubusercontent.com/25863377/28662750-8bd01612-72bb-11e7-9ff3-cde78be1cb14.jpg)

Not sure about the right mouse click. The user should be able to use the system without. It could be left in as an additional option to perform certain actions though. 

Any feedback is welcome! Feel free to start an issue about any element. :)

**Comment by rtripault** [comment]
@RinzeVanDerBrug yes, i removed the top navigation since (at least on desktop) we have more width and height available.
Adding its content to the side bar (but again, too much data in there). The entries are labelled 'user nav' & 'main nav' if you want to spot them.

Here is another demo with that manager theme in case that could bring some ideas : 

https://www.youtube.com/watch?v=mKRaqySqki4

* search results ("uber bar" filling most of the screen) + custom records (system settings in that case)
* no more right click for textual menus in grids (in favor of a "cog" icon... probably not a11y friendly tho)
* highlight of editable fields in grids (again, probably not a11y friendly as it works on "hover")
* "sticky" grid headers
* experiments to add custom tabs/actions (save/close...) when editing a resource
* famous and "beloved" hamburger icon to toggle the left sidebar (not visible on the video but i can also automatically collapse when a breakpoint is reached, for small screens)

**Comment by jcdm** [comment]
The single add button is nice, but has one draw-back on the surface - how does it know which context we'd like to add to? Does it make sense to have it where it is here for a single context site but then reduce the height and move it into the context title bar for multi-context sites?

While I don't like moving components between sites from a UI perspective, I like the simplicity of this but don't know how much sense it makes in keeping a multi-context site simple.

**Comment by jcdm** [comment]
And I like the vertical tree, but could we keep a second tab for the Elements? That would make it really easy to have everything the 'content editors' need in one place and the things that people working with site structure need in another place, but both right within reach.

**Comment by RinzeVanDerBrug** [comment]
@rtripault Thanks for your contribution, Romain. Will analyse your work!

**Comment by RinzeVanDerBrug** [comment]
@jcdm Thank you! I understand your question. I thought about the context problem. Two options to make it work with only one 'Add' button for all contexts:
**Option 1:** The created document will be automatically added to the context that's active (the one your currently working in).
**Option 2:** After you click 'Add resource' a pop-up window will ask you in what context you would like to place it (my preference). 

I like the elements on a second tab idea. :-)

**Comment by rthrash** [comment]
When you are in the Manager, the active context is Mgr. Perhaps the add action Cloud be three steps for multicontext sites:

1. Click to start the add
2. From the resulting drop down, choose what you want to add as mocked up
3. (New) a list of contexts slides over the drop down to click to choose the destination.

**Comment by rthrash** [comment]
@jcdm what if there were keyboard shortcuts to focus on the resources or the elements tree, or static anchor buttons/links at the top for quickly focusing/opening the relevant sections?  (Configurable per user/role per Romain's discussion of course!)

**Comment by jcdm** [comment]
Apart from the manager thing, the active context is also a bit of a misnomer - if I log in and 'Add' is the first thing I click it's not obvious to me as a user what I'm interacting with.

I think **Option 2** of the inserted step clarifying is the go. It can be skipped for single context sites. I still feel like this is a step back from one add per context like we have now, but no doubt I'll get used to it 👍🏻

But if I could right-click within a context to add like I can now as well, that would tick that box for me.

**Comment by jcdm** [comment]
@rthrash I'd like keyboard shortcuts AND for it to be a second tab. In my mind the second tab segments 'front of house' and 'behind the curtain' nicely but I'm aware that makes sense to me in my workflow but may not have universal appeal.

But if I could then jump straight to the snippets in the other tab with a keyboard shortcut and then back to the resource tree with another...happiest of days!

**Comment by labr1005** [comment]
Maybe

1. Bring back the triangles showing that a resource has children.
2. Make the icons more meaningful by adding "sub icons". Italic = unpublished always has to be explained to clients.
3. Add connecting lines to illustrate the hierarchy/depth.
4. This will be a hard one: Don't make a resource a container (icon) automatically when it has children. I never liked that. Maybe make that only adjustable through the resource's setting to show the client/editor "this is only a container for something (e.g. a Meta Navigation) and has no content itself". Or use this for the feature "#11153 Exclude resource alias from child URIs"?

Here's a quick draft:

![resource-tree](https://user-images.githubusercontent.com/1913250/28693862-21885e06-7327-11e7-88a2-9db620f38991.png)

**Comment by labr1005** [comment]
I'm not sure if I like the main "Add +" button.

#### "The created document will be automatically added to the context that's active (the one your currently working in)."

And when all contexts are minimized?

#### "After you click 'Add resource' a pop-up window will ask you in what context you would like to place it (my preference)."

Which basically only means 1 more action by the user (compared to now) that has no benefit and that's really not necessary - especially when there's only 1 context.

And what if I'm allowed to add weblinks to context A but not to context B?

I still would prefer something like this:

![modx---edit-page---tree-dropdown](https://user-images.githubusercontent.com/1913250/28717261-5499bf04-73a1-11e7-81c5-a073cebd019c.jpg)

There's no question here about where you're going to add something.

#### "Not sure about the right mouse click. The user should be able to use the system without. It could be left in as an additional option to perform certain actions though."

Not that this would be general, but 9 out of 10 clients of ours are not allowed to add resources to the root level. So they either add a child resource via right click or they add a new resource to a collection.

An alternative could be, to add a "+" button to each resource which would be the action "add a new child resource inside this resource" (not sure how to handle weblinks, symlinks, etc. and new resources on root level with this).

![modx---edit-page---tree-dropdown-hover-add](https://user-images.githubusercontent.com/1913250/28717271-63d3c744-73a1-11e7-9a55-09e60f94be32.jpg)

Here I also removed the outer container for contexts.

"mouse hover" would also mean first click on touch, navigate with arrow up/down, etc.

**Comment by rthrash** [comment]
Great suggestions @labr1005

**Comment by Lefthandmedia** [comment]
Maybe put the + button only in the active context and grey out the inactives contexts to make instinctively clear which one is active. And remove the refresh button for inactive context (less clutter)
![xdscreensnapz005](https://user-images.githubusercontent.com/3350108/35627059-073340c0-0698-11e8-9d1f-40b1bc894dd1.png)
![xdscreensnapz006](https://user-images.githubusercontent.com/3350108/35627064-0a7846f4-0698-11e8-914d-c8c7a8079108.png)
